src/test/java/commission/services/PaymentCalculatorServiceTest.java

```diff
@@ -0,0 +1,202 @@
+        when(saleDao.findById(3)).thenReturn(Optional.of(new Sale(3, 3000, OffsetDateTime.parse("2024-02-03T10:15:30+02:00"), 12)));
+        when(saleDao.findById(4)).thenReturn(Optional.of(new Sale(4, 1000, OffsetDateTime.parse("2024-03-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(5)).thenReturn(Optional.of(new Sale(5, 7000, OffsetDateTime.parse("2024-03-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(6)).thenReturn(Optional.of(new Sale(6, 14000, OffsetDateTime.parse("2024-04-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(7)).thenReturn(Optional.of(new Sale(7, 5000, OffsetDateTime.parse("2024-05-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(8)).thenReturn(Optional.of(new Sale(8, 10000, OffsetDateTime.parse("2024-06-22T15:36:48+02:00"), 13)));
+        when(saleDao.findById(10)).thenReturn(Optional.of(new Sale(10, 0, OffsetDateTime.parse("2024-12-10T10:15:30+02:00") , 20)));
+
+        when(commissionDao.findById(1)).thenReturn(Optional.empty());
+        when(commissionDao.findById(2)).thenReturn(Optional.of(new Commission(2, STRAIGHT, 200, 2)));
+        when(commissionDao.findById(3)).thenReturn(Optional.of(new Commission(3, GROSS_MARGIN, 120, 3)));
+        when(commissionDao.findById(4)).thenReturn(Optional.of(new Commission(4, TIERED, 100, 4)));
+        when(commissionDao.findById(5)).thenReturn(Optional.of(new Commission(5, TIERED, 1050, 5)));
+        when(commissionDao.findById(6)).thenReturn(Optional.of(new Commission(6, TIERED, 2800, 6)));
+        when(commissionDao.findById(7)).thenReturn(Optional.of(new Commission(7, TIERED, 750, 7)));
+        when(commissionDao.findById(8)).thenReturn(Optional.of(new Commission(8, TIERED, 1500, 8)));
+        when(commissionDao.findById(10)).thenReturn(Optional.of(new Commission(10, TIERED, 1, 10)));
+    }
+
+    @AfterEach
+    public void tearDown(){
+        reset(saleDao, commissionDao);
+    }
+
+    @Test
+    public void givenCompleteCommission_WhenSearchingForSaleById_ThenVerifyTheSearch(){
+        var result = new Commission(10, STRAIGHT, 200, 2);
+        paymentCalculatorService.completeCommission(result);
+
+        verify(saleDao, times(1)).findById(2);
+    }
+
+
+    @Test
+    public void givenCalculateCommission_WhenCalculatingWithPriceZero_ThenThrowError(){
+        var result =commissionDao.findById(10);
+        assertTrue(result.isPresent());
+        assertThrows(ArithmeticException.class, () -> paymentCalculatorService.calculateCommissionPayment(result.get()));
+        verify(saleDao, times(1)).findById(10);
+    }
+
+    @Test
+    public void givenCalculateCommissionPayment_WhenCalculatingWithTypeStraight_ThenConfirmTypeStraight(){
+        var result = commissionDao.findById(2);
+
+        assertTrue(result.isPresent());
+        paymentCalculatorService.calculateCommissionPayment(result.get());
+
+        assertEquals(result.get().commissionType(), STRAIGHT);
```

Comment:
You don't need to assert that your input data is correct. Assertions / verification is only used for output and interaction with mocks. This ties into my earlier point that you should define the test data in the test not via a mock, because then you can see what you are passing into the method being tested and don't need to assert the input data.

Reply:
That is very obvious indeed. So this tests are not really testing anything. 
Am I right in thinking that confirming the type is not really necessary, because that is defined by the user, but what matters is that the calculation and result is correct?

Reply:
It is defined by your input for the test so that is like writing something like this:
```
var number = 5;
//do test logic
assertEqual(5, number); // <- this you have already set yourself and it can't be changed by the test code
```

Reply:
In simple words, test logic, test code, and not my own input (obviously)